### PR TITLE
Fix bug to query header search attributes correctly in visibility

### PR DESCRIPTION
**Pull request by shijiesheng**
**What changed?**

* Change the format of header search attributes in visibility and added sanitization
* Revert https://github.com/uber/cadence/pull/6144/files

**Why?**

* search attributes cannot contain `.` and this assumption exists in multiple places in the code
* sqlparser library does not allow `-` in the column name in where clause. 

**How did you test it?**

unit test

**Potential risks**

This feature only exists in pre-releases and internally it's not rolled out. Additionally, this feature is also behind a feature flag so it's very safe. 

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/history/task/transfer_task_executor_base.go [hunk 1]
```diff
@@ -50,6 +51,8 @@ const (
 	defaultDomainName              = "defaultDomainName"
 )
 
+var nonAlphanumericRegex = regexp.MustCompile(`[^a-zA-Z0-9]+`)
```
Comment (taylanisikdemir) [inline service/history/task/transfer_task_executor_base.go:54]: it's better to not spread these visibility store constraints across the codebase. can you move `sanitizedHeaderKey` to common folder or somewhere in visibility folders?

File: tools/cli/admin_cluster_commands.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ import (
 
 // An indirection for the prompt function so that it can be mocked in the unit tests
 var promptFn = prompt
-var validSearchAttributeKey = regexp.MustCompile(`^[a-zA-Z][a-zA-Z_.-0-9]*$`)
+var validSearchAttributeKey = regexp.MustCompile(`^[a-zA-Z][a-zA-Z_0-9]*$`)
```
Comment (taylanisikdemir) [inline tools/cli/admin_cluster_commands.go:39]: same here. this regex should live close to other search attribute handling code in a central place.

File: service/history/task/transfer_task_executor_base.go (hunk 1 above)
Comment (shijiesheng) [reply service/history/task/transfer_task_executor_base.go]: make sense. Now it's in the common folder

File: service/history/task/transfer_task_executor_base.go [hunk 3]
```diff
@@ -30,6 +30,7 @@ import (
 	"github.com/uber/cadence/common"
 	"github.com/uber/cadence/common/backoff"
 	"github.com/uber/cadence/common/definition"
+	"github.com/uber/cadence/common/elasticsearch/validator"
```
Comment (taylanisikdemir) [inline service/history/task/transfer_task_executor_base.go:33]: elasticseach is a plugin. rest of the codebase shouldn't directly depend on this package. same search attribute constrationts apply to Pinot and other future visibility stores as well. so let's move this validator to `common/visibility` or somewhere generic like that

**Review by taylanisikdemir — APPROVED** [review]

File: service/history/task/transfer_task_executor_base.go (hunk 3 above)
Comment (shijiesheng) [reply service/history/task/transfer_task_executor_base.go]: done

**Pull request merged**